### PR TITLE
[core] feat: add MenuItem small modifier, extend MenuDivider width

### PR DESCRIPTION
**Pull request by adidahiya**
#### Checklist

- [ ] Includes tests
- [x] Update documentation

#### Changes proposed in this pull request:

- feat(`MenuItem`): add support for `small` size modifier prop
- feat(`MenuItem`): remove constraint that prevents both selected "tick" icon and `icon` prop from being shown at the same time
- fix(`MenuItem`): "selected" styles no longer extend "active" styles (this is mostly an internal change)

#### Reviewers should focus on:

No regressions in selected styling and possible CSS overrides consumers may be applying on `.bp5-menu-item.bp5-selected` elements.

### CODE REVIEW

**Comment by adidahiya** [comment]
<h3>Merge remote-tracking branch 'origin/develop' into ad/menu-item-styling</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/3962af74-17d6-437f-a979-5b1355cb3404/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/3962af74-17d6-437f-a979-5b1355cb3404/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/3962af74-17d6-437f-a979-5b1355cb3404/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/3962af74-17d6-437f-a979-5b1355cb3404/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

File: packages/core/src/components/menu/menuSection.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+/*
```
Comment (adidahiya) [inline packages/core/src/components/menu/menuSection.tsx]: I don't think this component is necessary. It makes the API for defining MenuItems look different from and much more verbose than the current API. This syntax change from JSX props -> JS object key-value pairs feels a bit weird and I don't want to make users refactor their code like this:

before:

```tsx
<MenuDivider title="Edit" />
<MenuItem icon="cut" text="Cut" label="⌘X" />
<MenuItem icon="duplicate" text="Copy" label="⌘C" />
<MenuItem icon="clipboard" text="Paste" label="⌘V" disabled={true} />
<MenuDivider title="Text" />
```

after:

```tsx
<MenuSection
    sectionTitle="Edit"
    items={[
        {
            icon: "cut",
            id: "cut",
            label: "⌘X",
            text: "Cut",
        },
        {
            icon: "duplicate",
            id: "duplicate",
            label: "⌘C",
            text: "Copy",
        },
        {
            disabled: true,
            icon: "clipboard",
            id: "clipboard",
            label: "⌘V",
            text: "Paste",
        },
    ]}
/>
```

File: packages/core/src/components/menu/_menu.scss [hunk 2]
```diff
@@ -69,6 +71,14 @@ Styleguide menu
...
   }
+
+  .#{$ns}-small & {
+    @include menu-item-small();
+  }
+}
+
+.#{$ns}-menu-no-padding {
+  padding: 0;
 }
```
Comment (adidahiya) [inline packages/core/src/components/menu/_menu.scss]: You don't need to create a new modifier class to make dividers take up full width. You can use negative margins to achieve the same result:

image

Also I would prefer not to support _both_ the current MenuDivider visual design _and_ a new variant where the divider takes up full width. I think we should be opinionated and just pick one. If we think full width dividers are the direction we want the visual design to go towards, then let's just change the styles to reflect that and make all MenuDividers full width.

**Comment by adidahiya** [comment]
<h3>Remove MenuSection and noPadding prop</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/ce72245d-0bf4-4e1c-a5a1-5741033d86da/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/ce72245d-0bf4-4e1c-a5a1-5741033d86da/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/ce72245d-0bf4-4e1c-a5a1-5741033d86da/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/ce72245d-0bf4-4e1c-a5a1-5741033d86da/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

**Comment by adidahiya** [comment]
There's a bug in the latest docs preview where the docs navigator menu items have an unexpected left padding: 
image

**Comment by adidahiya** [comment]
Also, it looks like `<Suggest>` items are not getting the "selected" tick icon: 
image

**Comment by adidahiya** [comment]
<h3>Fix Omnibar, Navigator, and select examples</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/5a211b95-c98e-47b2-bba8-70e4ccf95a5c/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/5a211b95-c98e-47b2-bba8-70e4ccf95a5c/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/5a211b95-c98e-47b2-bba8-70e4ccf95a5c/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/5a211b95-c98e-47b2-bba8-70e4ccf95a5c/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

File: packages/core/src/components/menu/_menu.scss [hunk 3]
```diff
@@ -29,6 +29,7 @@ Markup:
 </ul>
 
 .#{$ns}-large - Large size (only supported on <code>.#{$ns}-menu</code>)
+.#{$ns}-small - Snall size (only supported on <code>.#{$ns}-menu</code>)
```
Comment (adidahiya) [inline packages/core/src/components/menu/_menu.scss]: minor typo

```suggestion
.#{$ns}-small - Small size (only supported on <code>.#{$ns}-menu</code>)
```

**Review by CPerinet — APPROVED** [review]

File: packages/core/src/components/menu/menuItem.tsx [hunk 4]
```diff
@@ -263,6 +263,11 @@ export const MenuItem: React.FC<MenuItemProps> = React.forwardRef<HTMLLIElement,
             ...(disabled ? DISABLED_PROPS : {}),
             className: anchorClasses,
         },
+        isSelected ? (
+            <span className={Classes.MENU_ITEM_SELECTED_ICON}>
```
Comment (adidahiya) [inline packages/core/src/components/menu/menuItem.tsx]: can this class be applied directly to `<SmallTick />`? we might be able to collapse this and remove the extra container element.

File: packages/docs-app/src/examples/core-examples/menuItemExample.tsx [hunk 5]
```diff
@@ -33,33 +34,34 @@ export function MenuItemExample(props: ExampleProps) {
...
                 <BooleanOrUndefinedSelect
-                    disabled={!isSelectedOptionAvailable}
+                    disabled={!isSelectable}
                     label="Selected"
                     value={selected}
                     onChange={setSelected}
                 />
             </PropCodeTooltip>
-            <Switch label="Enable icon" checked={iconEnabled} onChange={handleBooleanChange(setIconEnabled)} />
+            <Switch label="Show custom icon" checked={iconEnabled} onChange={handleBooleanChange(setIconEnabled)} />
```
Comment (adidahiya) [inline packages/docs-app/src/examples/core-examples/menuItemExample.tsx]: "Custom icon" may not be the right terminology here. It creates ambiguity with the custom SVG icon shown elsewhere in this docs page.

File: packages/docs-app/src/examples/select-examples/suggestExample.tsx [hunk 6]
```diff
@@ -161,22 +161,35 @@ export class SuggestExample extends React.PureComponent<ExampleProps, SuggestExa
...
+                {...getFilmItemProps(film, props)}
+                roleStructure="listoption"
+                selected={film === this.state.selectedFilm}
+            />
+        );
+    };
+
     private renderInputValue = (film: Film) => film.title;
 
     private handleValueChange = (film: Film) => {
```
Comment (adidahiya) [inline packages/docs-app/src/examples/select-examples/suggestExample.tsx]: nit:

```suggestion
    private handleValueChange = (selectedFilm: Film) => {
```

**Review by adidahiya — COMMENTED** [review]
self-review

**Comment by adidahiya** [comment]
The click-and-hold active state style has changed:



this is a regression from existing behavior:

**Comment by adidahiya** [comment]
<h3>address self-review, update docs</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/680280cc-a88f-4d72-8136-0dd340d61f5a/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/680280cc-a88f-4d72-8136-0dd340d61f5a/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/680280cc-a88f-4d72-8136-0dd340d61f5a/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/680280cc-a88f-4d72-8136-0dd340d61f5a/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

**Comment by adidahiya** [comment]
There's also a regression in dark theme styles with the `bp5-active` modifier class:

image


it's supposed to look like this (from last release):

image

I'll push a commit to fix this regression and the one I noted in my last comment.

**Review by adidahiya — COMMENTED** [review]
LGTM :shipit:

**Comment by adidahiya** [comment]
<h3>fix :active styles</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/eab59606-7027-495c-9656-f9dadc0307bb/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/eab59606-7027-495c-9656-f9dadc0307bb/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/eab59606-7027-495c-9656-f9dadc0307bb/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/eab59606-7027-495c-9656-f9dadc0307bb/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

**Pull request merged**